zipline/utils/calendars/trading_calendar.py

```diff
@@ -677,8 +677,13 @@ def days_at_time(days, t, tz, day_offset=0):
     day_offset : int
         The number of days we want to offset @days by
     """
-    days = DatetimeIndex(days).tz_localize(None).tz_localize(tz)
+
+    # Offset days in UTC to avoid discontinuities due to DST, before
+    # localizing to tz.
+    days = DatetimeIndex(days).tz_localize(None).tz_localize('UTC')
```

Comment:
Yes, I'm adding the test on my branch (building the test framework for the CME calendar (and others) there).

It's fixing this problem:
```
(Pdb++) calendar.schedule.loc['2004-04-05']
market_open    2004-04-03 23:01:00
market_close   2004-04-05 22:00:00
Name: 2004-04-05 00:00:00+00:00, dtype: datetime64[ns]
```